packages/back-end/src/services/features.ts

```diff
@@ -393,3 +419,37 @@ export function getApiFeatureObj(
 
   return featureRecord;
 }
+
+export function getNextScheduledUpdate(
+  envSettings: Record<string, FeatureEnvironment>
+) {
+  if (!envSettings) {
+    return null;
+  }
+
+  const dates: string[] = [];
+
+  if (envSettings) {
+    for (const env in envSettings) {
+      const rules = envSettings[env].rules;
+
+      rules.forEach((rule: FeatureRule) => {
+        if (rule.scheduleRules) {
```

Comment:
I think I can simplify this by spreading in the scheduleRules array into the dates array, rather than just looping through it.